### PR TITLE
refactor: use Screen Component in NWFY screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

This PR is my favourite kind of PRs 😄 a PR where you **delete more code than you write**.

- What's special about this PR and why was this **PR** important?
This PR is an example of how we can manage to solve what I consider the most annoying issue I hate when building scrollviews in RN, it's nested scrollviews. By [exposing the hook as a plugin](https://github.com/artsy/eigen/pull/9763), we can manage to hook it to our custom ScrollViews (InifiniteGrid, Artwork Grids, Lists....) from anywhere and not need any nesting

_PS: I will talk to design about moving the # of artworks. It shouldn't be a blocker_

https://github.com/artsy/eigen/assets/11945712/8fdcd4e5-0dd6-44e1-b10a-c3c88e51f6c2

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- refactor: NWFY to use Screen component - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by gkartalis — COMMENTED** [review]
awesome!!! 👏

File: src/app/Components/ArtworkGrids/MasonryInfiniteScrollArtworkGrid.tsx [hunk 1]
```diff
@@ -92,6 +98,11 @@ export const MasonryInfiniteScrollArtworkGrid: React.FC<MasonryInfiniteScrollArt
       ListFooterComponent={
         <MasonryListFooterComponent shouldDisplaySpinner={shouldDisplaySpinner} />
       }
+      onScroll={rest.onScroll}
     />
   )
 }
+
+const AnimatedFlashList = Animated.createAnimatedComponent(
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Components/ArtworkGrids/MasonryInfiniteScrollArtworkGrid.tsx:106]: of course, with pleasure. Good comment

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (refactor: NWFY to use Screen component - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cfaffa5a7fdb3d71a1778f6b01d9e16adee9c792
</p>

**Review by olerichter00 — APPROVED** [review]
Amazing 🌟

**Comment by MounirDhahri** [comment]
Let's goooo

**Pull request merged**